src/graph_engine.jl

```diff
@@ -258,11 +258,12 @@ getname(labels::ResizableArray{T, V, N} where {T <: NodeLabel, V, N}) = getname(
 iterate(label::NodeLabel) = (label, nothing)
 iterate(label::NodeLabel, any) = nothing
 
-to_symbol(label::NodeLabel) = Symbol(String(label.name) * "_" * string(label.global_counter))
+to_symbol(label::NodeLabel) = to_symbol(label.name, label.global_counter)
+to_symbol(name::D, index::Int) where {D} = Symbol(String(name) * "_" * string(index))
```

Comment:
```suggestion
to_symbol(name::Any, index::Int) = Symbol(string(name, "_", index))
```
this should further reduce allocations
also, maybe not a part of this PR, but `Symbol` by itself is quite expensive to create, maybe we could use `String` directly?